internal/controller/resilient_deployment_workflow.go

```diff
@@ -0,0 +1,623 @@
+
+	kObj := GetKuadrantFromTopology(topology)
+	if kObj == nil {
+		logger.V(level).Info("kuadrant resource has not being created yet.")
+		return nil
+	}
+	lObj := GetLimitadorFromTopology(topology)
+	if lObj == nil {
+		logger.V(level).Info("limitador resource has not being created yet.")
+		return nil
+	}
+
+	if r.startCleanup(kObj) {
+		logger.V(level).Info("Reconciling Rate Limiting resilient configuration", "status", "started")
+		defer logger.V(level).Info("Reconciling Rate Limiting resilient configuration cleanup", "status", "completed")
+
+		err := r.cleanupLimitadorDeployment(ctx, topology, logger)
+		if err != nil {
+			logger.V(level).Info("failed to cleanup limitador deployment", "status", "error", "error", err)
+			state.Store(ResilienceError, true)
+			return nil
+		}
+
+		err = r.cleanupLimitador(ctx, logger, lObj)
+		if err != nil {
+			logger.V(level).Info("failed to cleanup limitador resource", "status", "error", "error", err)
+			state.Store(ResilienceError, true)
+			return nil
+		}
+
+		return nil
+	}
+
+	if r.startReconcile(kObj) {
+		logger.V(level).Info("Reconciling Rate Limiting resilient configuration creation", "status", "started")
+		defer logger.V(level).Info("Reconciling Rate Limiting resilient configuration creation", "status", "completed")
+
+		err := r.configureLimitador(ctx, logger, lObj)
+		if err != nil {
+			logger.V(level).Info("failed to configure limitador deployment", "status", "error", "error", err)
+			state.Store(ResilienceError, true)
+			return nil
+		}
+
+		err = r.configureLimitadorDeployment(ctx, topology, logger)
+		if err != nil {
+			logger.V(level).Info("failed to configure limitador deployment", "status", "error", "error", err)
+			state.Store(ResilienceError, true)
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (r *ResilienceRateLimitingReconciler) configureLimitador(ctx context.Context, logger logr.Logger, lObj *limitadorv1alpha1.Limitador) error {
+	update := false
+	lObj.Spec.Replicas = ptr.To(LimitadorReplicas)
+
+	if !limitadorPDBIsConfigured(lObj) {
+		if lObj.Spec.PodDisruptionBudget == nil {
+			logger.Info("setting the pdb", "status", "working")
+			lObj.Spec.PodDisruptionBudget = &limitadorv1alpha1.PodDisruptionBudgetType{
+				MaxUnavailable: &intstr.IntOrString{IntVal: LimitadorPDB},
+			}
+			update = true
+		}
+	}
+
+	if !limitadorResourceRequestsIsConfigured(lObj) {
+		logger.Info("setting the Resource Request", "status", "working")
+
+		if lObj.Spec.ResourceRequirements == nil {
+			lObj.Spec.ResourceRequirements = &corev1.ResourceRequirements{}
+		}
+
+		if lObj.Spec.ResourceRequirements.Requests == nil {
+			lObj.Spec.ResourceRequirements.Requests = make(corev1.ResourceList)
+		}
+
+		if lObj.Spec.ResourceRequirements.Requests.Cpu().Value() == 0 {
+			cpu, err := resource.ParseQuantity(Resource10m)
+			if err != nil {
+				logger.Error(err, "failed to parse resurce cpu string", "status", "error")
```

Comment:
should we return the error here rather than attempt to update ? I see it is highly unlikely given we are parsing a constant but it seems odd to just log and attempt to update anyway?